storage/innobase/ddl/ddl0builder.cc

```diff
@@ -2045,19 +2047,17 @@ dberr_t Builder::finish() noexcept {
   }
 
   dberr_t err{DB_SUCCESS};
-
-  if (get_error() != DB_SUCCESS || !m_ctx.m_online) {
+  if (get_error() == DB_SUCCESS) {
+    bool apply_log = true;
     /* Do not apply any online log. */
-  } else if (m_ctx.m_old_table != m_ctx.m_new_table) {
-    ut_a(!m_index->online_log);
-    ut_a(m_index->online_status == ONLINE_INDEX_COMPLETE);
-
-    auto observer = m_ctx.m_trx->flush_observer;
-    observer->flush();
-
-  } else {
-    err = finalize();
-
+    if (!m_ctx.m_online) {
+      apply_log = false;
```

Comment:
this should be true right? with EXCLUSIVE, m_online is false

Reply:
> this should be true right? with EXCLUSIVE, m_online is false

yes. with EXCLUSIVE m_online is false.

Reply:
ok, apply_log doesn't have effect on flush observer flush and writing redo. So this is fine?

Reply:
yeah. we discussed that we need to flush if we have used btr_blk. so that's why we would be flushing in all three cases.  old_table!=new_table, not online and  when online.

Reply:
right rahul, makes sense